src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,633 @@
+constexpr int nimptblgrow_total = -nimptblgrow_mind + nimptblgrow_maxd + 1;
+const int nrainsvmax = 50; // maximum bin number for rain
+const int naerosvmax = 51; //  maximum bin number for aerosol
+const int maxd_aspectype = 14;
+
+KOKKOS_INLINE_FUNCTION
+void modal_aero_bcscavcoef_get(
+    const int imode, const bool isprx_kk, const Real dgn_awet_imode_kk, //& ! in
+    const Real dgnum_amode_imode,
+    const Real scavimptblvol[nimptblgrow_total][AeroConfig::num_modes()],
+    const Real scavimptblnum[nimptblgrow_total][AeroConfig::num_modes()],
+    Real &scavcoefnum_kk, Real &scavcoefvol_kk) {
+
+  // !-----------------------------------------------------------------------
+  // ! compute impaction scavenging removal amount for aerosol volume and number
+  // !-----------------------------------------------------------------------
+
+  // @param [in]  imode mode index
+  // @param [in]  isprx_kk if there is precip
+  // @param [in]  dgn_awet_imode_kk ! wet aerosol diameter of mode imode at
+  // elevation kk [m]
+  // @param [out] scavcoefnum scavenging removal for aerosol number [1/h]
+  // @param [out] scavcoefvol scavenging removal for aerosol volume [1/h]
+
+  // NOTE: original FORTRAN function has two internal loops: kk and icol.
+  // We removed these loops. Hence, the inputs/outputs of
+  // modal_aero_bcscavcoef_get are reals at kk , icol locations
+
+  const Real zero = 0;
+  const Real one = 1;
+  // BAD CONSTANT
+  const Real dlndg_nimptblgrow = haero::log(1.25);
+  if (isprx_kk) {
+    // With precipitation
+    // interpolate table values using log of
+    // (actual-wet-size)/(base-dry-size) ratio of wet and dry aerosol diameter
+    // [fraction]
+    const Real wetdiaratio = dgn_awet_imode_kk / dgnum_amode_imode;
+    // Note: indexing in scavimptblnum and scavimptblvol
+    // Fortran : [-7,12]
+    // C++ :  [0,19]
+    // Therefore, -7 (Fortran) => 0 (C++), or jgrow => jgrow - nimptblgrow_mind;
+    // Here, we are assuming that nimptblgrow_mind is negative
+    Real scavimpvol, scavimpnum = zero;
+    // BAD CONSTANT
+    if (wetdiaratio >= 0.99 && wetdiaratio <= 1.01) {
+      // 8th position: Fortran (0) C++(7 or -nimptblgrow_mind)
+      scavimpvol = scavimptblvol[-nimptblgrow_mind][imode];
+      scavimpnum = scavimptblnum[-nimptblgrow_mind][imode];
+    } else {
+      Real xgrow = haero::log(wetdiaratio) / dlndg_nimptblgrow;
+      int jgrow = int(xgrow); // get index jgrow
+      if (xgrow < zero) { // // adjust jgrow appropriately if xgrow is negative
+        jgrow = jgrow - 1;
+      }
+      // bound jgrow within max and min values
+      if (jgrow < nimptblgrow_mind) {
+        jgrow = nimptblgrow_mind;
+        xgrow = jgrow;
+      } else {
+        jgrow = haero::min(jgrow, nimptblgrow_maxd - 1);
+      }
+      // compute factors for interpolating impaction scavenging removal amounts
+      const Real dumfhi = xgrow - jgrow;
+      const Real dumflo = one - dumfhi;
```

Comment:
@tangshuaiqi had a similar question. Here is @jeff-cohere 's reply:
> 0 and 1 are special numbers, often used to initialize things. But to me this is a matter of taste.

In addition to Jeff's comment. I also use one for 1 or zero for 0 to avoid issues between single and double precision.    

The concept of "magic number" seems to apply to constant value such as $\pi$.